Kernel/Storage/AHCIPort.h

```diff
@@ -50,6 +51,21 @@ class AHCIPort : public RefCounted<AHCIPort> {
     friend class AHCIPortHandler;
     friend class SATADiskDevice;
 
+private:
+    class ScatterList : public RefCounted<ScatterList> {
+    public:
+        static NonnullRefPtr<ScatterList> create(AsyncBlockDeviceRequest&, NonnullRefPtrVector<PhysicalPage> allocated_pages, size_t device_block_size);
+        NonnullRefPtr<AnonymousVMObject> vmobject() const { return m_vm_object; }
```

Comment:
Since the `VMObject` is owned by the `ScatterList`, let's return it as a `const VMObject&` or `VMObject&`.